### ISSUE TITLE
Trouble changing project name

### ISSUE THREAD

**Issue by vizziv** [srid/ema-docs#11]
I've cloned the template repo and got the demo generally working. But when I tried to change one or both of the following, I get an error:

- the name of the `.cabal` file (`ema-template.cabal` to `my-project.cabal`)
- the project name in the `flake.nix` file (`name = "ema-template"` to `name = "my-project"`)

The error is

    cabal2nix: nix-prefetch-url: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)

This error occurs whether I change just the cabal file's name, just the line in the flake file, or both.

I also tried the aforementioned name changes along with various other changes, e.g. replacing `ema-template` with `my-project` in `hie.yaml` and other places, but those didn't seem to have a critical effect. In particular, changing *every* instance of `ema-template` in the repo to `my-project` didn't help.

In case it matters: my machine is macOS 10.15.7. I'm also a total Nix newbie.

**Comment by srid** [comment]
I think you should `git add` and `git commit` your changes, because Nix Flakes sort of ignores any files (such as the new cabal file) not in Git index.

We should add this to the README I suppose.

**Comment by vizziv** [comment]
So, I got this working... but it required a strange step.

Adding and committing with the new Cabal file name made progress: things like `nix build` and `nix-shell` worked. But the dev server (`bin/run`) threw a different error.

    ./ema-template.cabal: getModificationTime:getFileStatus: does not exist (No such file or directory)

What ultimately fixed that error was *changing the license field in the Cabal file*. (I did this unrelatedly, because the `LICENSE` file and the license field didn't match, and was surprised to see it fix the problem!) After a successful dev server run, I could change the license back and it still worked.

To confirm that this was actually the thing that worked, I tried changing the project's name again. Changing top-level Cabal metadata (license, author, etc.) was the step that made the difference.
